### DISCUSSION TITLE
Changing transition matrix for every time step

### DISCUSSION THREAD

**Discussion by anyperez** [RangeShifter/RangeshiftR-tutorials#3] [Q&A] [answered]
Hi everyone,

I'm doing a stage-structured model using the R package (RangeShiftR) and I would like to change the survival rates on the transition matrix for every time step of the simulation, picking up a random value from a specific range for every stage. Is it possible? When I use runif() as one of the entries of the transition matrix (see example below), it seems like it only takes one random value at the beginning of the simulation but uses the same matrix for every time step.

E.g. for juvenile survival rate:
stg <- StageStructure(Stages = 4, TransMatrix = matrix(c(0, runif(1, 0.34, 0.43),0,0,0,.72,.18,0,.35,0,.92,.05,.05,0,0,.71),nrow = 4), MaxAge = 50, MinAge = c(0,0,6,26))

I appreciate your help on this matter

**Comment by damariszurell** [comment, marked as answer]
Hi there, currently there is no option for varying the survival rates in each time step. We have it on our list for future extensions though. For such a feature, we need to change the underlying C++ code. Currently, it is only possible with a work-around based using user-define initialisation files. Say you want to run the model for time step t1 and then change the survival rates for time step t2. Then, you can use the population output from t1 to define the initalisation file in t2, change the survival rates in the stage structure module, update the Demography module, and define a new parameter master and simulation with new initialisiation and new demography module. It is a bit tedious though.

**Reply by anyperez** [reply]
Hi, thanks for your answer. So I guess it isn't possible either to incorporate catastrophes every now and then along the simulation, right?

**Reply by damariszurell** [reply]
Hi there, you can account for catastrophes using the dynamic landscapes. For example, setting the suitability of the landscape lower for certain time steps.
